src/librustdoc/html/static/main.js

```diff
@@ -1837,14 +1851,19 @@ function defocusSearchBar() {
                 var crateSize = 0;
 
                 searchWords.push(crate);
-                searchIndex.push({
+                var crateRow = {
                     crate: crate,
                     ty: 1, // == ExternCrate
                     name: crate,
                     path: "",
                     desc: rawSearchIndex[crate].doc,
+                    parent: undefined,
                     type: null,
-                });
+                    id: "",
```

Comment:
Since you set it just after, maybe better to give it a `null` value instead or not declare it at all? (Simple suggestion, not sure if that changes anything)

Reply:
It’s the same reason as above. I don’t want to change the field’s shape after I construct it.